src/control/controlobjectscript.cpp

```diff
@@ -1,9 +1,11 @@
+#include "control/controlobjectscript.h"
+
 #include <QtDebug>
 
-#include "control/controlobjectscript.h"
+#include "controllers/controllerdebug.h"
 
 ControlObjectScript::ControlObjectScript(const ConfigKey& key, QObject* pParent)
-        : ControlProxy(key, pParent, ControlFlag::NoAssertIfMissing) {
+        : ControlProxy(key, pParent, ControllerDebug::shouldAssertForInvalidControlObjects()) {
```

Comment:
> Please explain under which conditions we crash.

This is trivially discoverable by reading the implementation of this simple function. If you can suggest a better name for the function, then do so, but again, comments that simply repeat what the code already says are noise.

Reply:
I can't follow your explaination.
If a script calls getValue for a non existing CO, Mixxx crashed now.
That must not happen, because we consider a script as user data.
Please revert.

Reply:
> If a script calls getValue for a non existing CO, Mixxx crashed now.
> That must not happen, because we consider a script as user data.

Usually I'd agree, but here this behavior was requested explicitly by passing the `--controllerDebug` flag to Mixxx.
If you disagree, let's continue this discusson on https://github.com/mixxxdj/mixxx/pull/3056 instead.

Reply:
You responds is to an outdated message.
The latest message is few messages above.

Here it is again:


> That sound reasonable.
Unfortunately I had hard times to understand that from the code.
I think this can be solved by a comment or a better function name.

> From a distance I think it is that the root cause is that "shouldAssertForInvalidControlObjects = ControlFlag::None" read "Like Don't Assert". So I get hooked to the inverted logic.

> Instead the function should become a bool.
And than we can do here more visible.
shouldAssertForInvalidControlObjects() ? ControlFlag::None : ControlFlag::AllowMissingOrInvalid

> In addition ControlFlag::None can become ControlFlag::AssertValid or something